src/devices/src/virtio/device.rs

```diff
@@ -5,20 +5,46 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the THIRD-PARTY file.
 
-use std::sync::{atomic::AtomicUsize, Arc};
+use std::sync::{atomic::AtomicUsize, atomic::Ordering, Arc};
 
 use super::{ActivateResult, Queue};
 use crate::virtio::AsAny;
 use utils::eventfd::EventFd;
 use vm_memory::GuestMemoryMmap;
 
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+
 /// Enum that indicates if a VirtioDevice is inactive or has been activated
 /// and memory attached to it.
 pub enum DeviceState {
     Inactive,
     Activated(GuestMemoryMmap),
 }
 
+/// State for saving the generic fields of a VirtioDevice.
+#[derive(Debug, Versionize)]
+pub struct VirtioDeviceState {
+    pub avail_features: u64,
+    pub acked_features: u64,
+    pub device_type: u32,
+    pub queues: Vec<Queue>,
+    pub interrupt_status: AtomicUsize,
```

Comment:
This can just be `usize`, the state doesn't need to worry about synchronization.

Reply:
Done.